lib/EnzymeCore/src/EnzymeCore.jl

```diff
@@ -245,18 +349,47 @@ const ReverseHolomorphicWithPrimal = ReverseMode{true,false,DefaultABI, true, fa
 @inline clear_runtime_activity(::ReverseMode{ReturnPrimal,RuntimeActivity,ABI,Holomorphic,ErrIfFuncWritten}) where {ReturnPrimal,RuntimeActivity,ABI,Holomorphic,ErrIfFuncWritten} = ReverseMode{ReturnPrimal,false,ABI,Holomorphic,ErrIfFuncWritten}()
 
 """
-    struct ReverseModeSplit{ReturnPrimal,ReturnShadow,RuntimeActivity,Width,ModifiedBetween,ABI} <: Mode{ABI,ErrIfFuncWritten,RuntimeActivity}
+    struct ReverseModeSplit{
+        ReturnPrimal,
+        ReturnShadow,
+        Width,
+        RuntimeActivity,
+        ModifiedBetween,
+        ABI,
+        ErrFuncIfWritten
+    } <: Mode{ABI,ErrIfFuncWritten,RuntimeActivity}
+
+Subtype of [`Mode`](@ref) for split reverse mode differentiation, to use in [`autodiff_thunk`](@ref) and variants.
 
-Reverse mode differentiation.
-- `ReturnPrimal`: Should Enzyme return the primal return value from the augmented-forward.
-- `ReturnShadow`: Should Enzyme return the shadow return value from the augmented-forward.
-- `RuntimeActivity`: Should Enzyme differentiate with runtime activity on (default off).
-- `Width`: Batch Size (0 if to be automatically derived)
-- `ModifiedBetween`: Tuple of each argument's modified between state (true if to be automatically derived).
+# Type parameters
+
+- `ReturnShadow`: whether to return the shadow return value from the augmented-forward.
+- `Width`: batch size (pick `0` to derive it automatically)
+- `ModifiedBetween`: `Tuple` of each argument's "modified between" state (pick `true` to derive it automatically).
+- other parameters: see [`ReverseMode`](@ref)
+
+!!! warning
+    The type parameters of `ReverseModeSplit` are not part of the public API and can change without notice.
+    Please use one of the following concrete instantiations instead (combined with setter functions like [`set_runtime_activity`](@ref)): 
+    - [`ReverseSplitNoPrimal`](@ref)
+    - [`ReverseSplitWithPrimal`](@ref)
```

Comment:
also the helper functions here.

including also ReverseSplitModified and ReverseSplitWidth

Reply:
Shouldn't all helper functions be lowercase or uppercase? Those are easy to confuse with actual structs

Reply:
No disagreement, but when I asked folks if those should be changed for the breaking change I didn’t see any feedback on.

So they will remain as they are (they significantly predate the other helper functions).

We can try to change in a future breaking release but they should still be referenced presently (for example you didn’t know about the existing width one!)

Reply:
According to Julia's interpretation of semantic versioning, those helper functions were never documented so they were never public in the first place. Thus you don't need a breaking release to change them.
That is precisely why I go to the trouble of adding docstrings everywhere, so that we get a clearer picture of what is public and what is not from now on.

Reply:
In this context they were public since we told folks to use them. I agree it would've been nice to have had docstrings too -- but in this context I don't think they can be changed til next major bump

Reply:
@vchuravy thoughts?

Reply:
in any case this is somewhat moot now per https://github.com/EnzymeAD/Enzyme.jl/pull/1898

Reply:
Totally happy to come up with a good unified design for the helper functions and mark deprecations of the old ones to new choices, but the time for breaking changes that make it in to 0.13 has already passed.

Feel free to open a PR/issue to start designing their successor, but if we're adding docs, we need to document the functions we have now.

Reply:
Sounds good, I'll finish documenting the existing stuff in this PR tomorrow

Reply:
same comment to list/reference the helper functions here?

Reply:
done